RecoLocalCalo/HGCalRecAlgos/src/HGCalImagingAlgo.cc

```diff
@@ -457,39 +486,39 @@ void HGCalImagingAlgo::shareEnergy(const std::vector<KDNode>& incluster,
   const double toleranceScaling = std::pow(std::max(1.0,seeds.size()-1.0),2.0);
   std::vector<Point> prevCentroids;
   std::vector<double> frac(seeds.size()), dist2(seeds.size());
-  while( iter++ < iterMax && diff > stoppingTolerance*toleranceScaling ) {    
+  while( iter++ < iterMax && diff > stoppingTolerance*toleranceScaling ) {
     for( unsigned i = 0; i < incluster.size(); ++i ) {
       const Hexel& ihit = incluster[i].data;
       double fraction(0.0), fracTot(0.0), d2(0.0);
       for( unsigned j = 0; j < seeds.size(); ++j ) {
 	fraction = 0.0;
-	d2 = ( std::pow(ihit.x - centroids[j].x(),2.0) + 
-	       std::pow(ihit.y - centroids[j].y(),2.0) + 
+	d2 = ( std::pow(ihit.x - centroids[j].x(),2.0) +
+	       std::pow(ihit.y - centroids[j].y(),2.0) +
```

Comment:
(ihit.y - centroids[j].y())*(ihit.y - centroids[j].y()) is more efficient than using pow(...,2)
The same for the other lines that define d2